code/game/objects/structures/crates_lockers/closets/statue.dm

```diff
@@ -1,146 +1,106 @@
-			if(H.gender == "female")
-				icon_state = "human_female"
-		else if(ismonkey(L))
-			name = "statue of a monkey"
-			icon_state = "monkey"
-		else if(iscorgi(L))
-			name = "statue of a corgi"
-			icon_state = "corgi"
-			desc = "If it takes forever, I will wait for you..."
-
-	if(obj_integrity == 0) //meaning if the statue didn't find a valid target
-		qdel(src)
-		return
-
-	START_PROCESSING(SSobj, src)
+		START_PROCESSING(SSobj, src)
 	..()
-	icon = L.icon
-	icon_state = L.icon_state
-	overlays = L.overlays
-	color = list(rgb(77,77,77), rgb(150,150,150), rgb(28,28,28), rgb(0,0,0))
 
-/obj/structure/closet/statue/process()
+/obj/structure/statue/petrified/process()
+	if(!petrified_mob)
+		STOP_PROCESSING(SSobj, src)
 	timer--
-	for(var/mob/living/M in src) //Go-go gadget stasis field
-		M.setToxLoss(intialTox)
-		M.adjustFireLoss(intialFire - M.getFireLoss())
-		M.adjustBruteLoss(intialBrute - M.getBruteLoss())
-		M.setOxyLoss(intialOxy)
-		M.Stun(1) //So they can't do anything while petrified
+	petrified_mob.Stun(2) //So they can't do anything while petrified
 	if(timer <= 0)
-		dump_contents()
 		STOP_PROCESSING(SSobj, src)
 		qdel(src)
 
-/obj/structure/closet/statue/dump_contents()
+/obj/structure/statue/petrified/contents_explosion(severity, target)
+	return
+
+/obj/structure/statue/petrified/handle_atom_del(atom/A)
+	if(A == petrified_mob)
+		petrified_mob = null
+
+/obj/structure/statue/petrified/Destroy()
 
 	if(istype(src.loc, /mob/living/simple_animal/hostile/statue))
 		var/mob/living/simple_animal/hostile/statue/S = src.loc
-		src.loc = S.loc
+		forceMove(S.loc)
 		if(S.mind)
-			for(var/mob/M in contents)
-				S.mind.transfer_to(M)
-				M.Weaken(5)
-				M << "<span class='notice'>You slowly come back to your senses. You are in control of yourself again!</span>"
-				break
+			if(petrified_mob)
+				S.mind.transfer_to(petrified_mob)
+				petrified_mob.Weaken(5)
+				petrified_mob << "<span class='notice'>You slowly come back to your senses. You are in control of yourself again!</span>"
 		qdel(S)
 
-
 	for(var/obj/O in src)
-		O.loc = src.loc
-
-	for(var/mob/living/M in src)
-		M.forceMove(loc)
-		M.disabilities -= MUTE
-		M.take_overall_damage((M.health - obj_integrity + 100)) //any new damage the statue incurred is transfered to the mob
-		M.faction -= "mimic"
-		M.reset_perspective(null)
-
-/obj/structure/closet/statue/take_contents()
-	return
-
-/obj/structure/closet/statue/open()
-	return
-
-/obj/structure/closet/statue/take_contents()
-	return
-
-/obj/structure/closet/statue/open()
-	return
-
-/obj/structure/closet/statue/insert()
-	return
-
-/obj/structure/closet/statue/close()
-	return
-
-/obj/structure/closet/statue/toggle()
-	return
-
-/obj/structure/closet/statue/MouseDrop_T()
-	return
-
-/obj/structure/closet/statue/relaymove()
-	return
-
-/obj/structure/closet/statue/attack_hand()
-	return
-
-/obj/structure/closet/statue/verb_toggleopen()
-	return
-
-/obj/structure/closet/statue/update_icon()
-	return
```

Comment:
Here's your clue that this thing should never have been a closet in the first place.